lib/server/services/files.js

```diff
@@ -0,0 +1,136 @@
+'use strict';
+
+const errors = require('storj-service-error-types');
+const log = require('../../logger');
+
+const { DELETING_FILE_MESSAGE } = require('../queues/messageTypes');
+
+// eslint-disable-next-line complexity
+async function removeFile(
+  storage,
+  { idBucket, userEmail, idFile },
+  { beforePointerIsRemoved = async () => { } }) {
+  const { Bucket, BucketEntry, Frame, Pointer } = storage.models;
+
+  if (!Bucket || !BucketEntry || !Frame || !Pointer) {
+    throw new errors.InternalError('Missing required storage models');
+  }
+
+  if (!idBucket || !userEmail || !idFile) {
+    throw new errors.InternalError('Missing required params');
+  }
+
+  try {
+    const bucket = await Bucket.findOne({ _id: idBucket });
+
+    if (!bucket) {
+      throw new errors.NotFoundError('Bucket not found');
+    }
+
+    if (bucket.user !== userEmail) {
+      throw new errors.ForbiddenError();
+    }
+
+    const bucketEntry = await getBucketEntry(BucketEntry, { idBucket: bucket._id, idFile });
+
+    if (!bucketEntry) {
+      throw new errors.NotFoundError('File not found');
+    }
+
+    const frame = await Frame.findOne({ _id: bucketEntry.frame.id });
+
+    if (!frame) {
+      log.error('Frame %s not found for file %s', bucketEntry.frame.id, bucketEntry._id);
+      await bucketEntry.remove();
```

Comment:
you could return the promise directly instead of awaiting and then returning: 
`return bucketEntry.remove();`

Reply:
fixed